src/Symfony/Component/Messenger/Worker.php

```diff
@@ -174,6 +175,8 @@ private function handleMessage(Envelope $envelope, ReceiverInterface $receiver,
 
     public function stop(): void
     {
+        $this->dispatchEvent(new WorkerStoppedEvent());
```

Comment:
Should this actually go at the end of the `run()` function? If it's here, is it possible this event is dispatched, and *then* one more message finishes being handled?

Reply:
Of course you're right, moved to the end of `run()`, thanks